### DISCUSSION TITLE
Question: Could OpenCrx provide functionality of a business directory in the following way?

### DISCUSSION THREAD

**Discussion by javac9** [opencrx/opencrx-documentation#18] [Q&A]
Originally asked here: [https://sourceforge.net/p/opencrx/discussion/329843/thread/cb2f765598/](https://sourceforge.net/p/opencrx/discussion/329843/thread/cb2f765598/)

It is clear that OpenCrx supports multiple segments. In a single segment, user can create Accounts (be it Legal entities, Contacts, etc.) and create relationships between them (assign certain contacts to appropriate company, ...).

Now, I am wondering if the following scenario would work:
Segment: Standard, no additional segments within the installation

Let's assume there is 100 "legal entity" accounts. Each of them was created by its real-world owner (which also has its own account of type Contact and is assigned as for example Founder to that legal entity).

Let's assume I am one of those 100 company owners. Do I get to see all the details of other companies, because in this scenario, all accounts have different creator? Do I see their contact data (addresses, phone numbers, their assigned contacts...)?

Does "Browse access level" setting item under "System" tab have something to do about this visibility question? I guess if some of those accounts all created by different users would be grouped together - they would be visible to each other, but is there a way to enable complete visibility without doing anything special?

**Comment by wfro64** [comment]
In a default setup all data is visible to all users since newly created objects are assigned to the PrincipalGroup 'Users' and all users are granted access to objects owned by 'Users'. In order to separate customers in a single segment you have to:

* Create a PrincipalGroup 'G.C' for each customer C
* Grant all users of this customer access to 'G.C'
* Set the default PrincipalGroup for these users to 'G.C'
* Set the owning group of all objects of customer C to 'G.C'

The default behaviour of openCRX supports this pattern (e.g. composite objects inherit the owning groups of their parent objects, primary principal group of user, ...). In general, however, custom logic is necessary.
